util/logging.go

```diff
@@ -0,0 +1,242 @@
+/*
+ * Copyright 2018 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+// Package util provides ancillary functionality to go-vcloud-director library
+// logging.go regulates logging for the whole library.
+// See LOGGING.md for detailed usage
+package util
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+const (
+	// Name of the environment variable that enables logging
+	envUseLog = "GOVCD_LOG"
+
+	// Name of the environment variable with the log file name
+	envLogFileName = "GOVCD_LOG_FILE"
+
+	// Name of the environment variable with the screen output
+	envLogOnScreen = "GOVCD_LOG_ON_SCREEN"
+
+	// Name of the environment variable that enables logging of passwords
+	envLogPasswords = "GOVCD_LOG_PASSWORDS"
+
+	// Name of the environment variable that enables logging of HTTP requests
+	envLogSkipHttpReq = "GOVCD_LOG_SKIP_HTTP_REQ"
+
+	// Name of the environment variable that enables logging of HTTP responses
+	envLogSkipHttpResp = "GOVCD_LOG_SKIP_HTTP_RESP"
+
+	// Name of the environment variable that limits the payload size
+	envLogMaxPayloadSize = "GOVCD_MAX_PAYLOAD_SIZE"
+
+	// Name of the environment variable that limits the result size
+	envLogMaxMaxResultSize = "GOVCD_MAX_RESULT_SIZE"
+)
+
+var (
+	// Name of the log file
+	// activated by GOVCD_LOG_FILE
+	ApiLogFileName string = "go-vcloud-director.log"
+
+	// Globally enabling logs
+	// activated by GOVCD_LOG
+	EnableLogging bool = false
+
+	// Enable logging of passwords
+	// activated by GOVCD_LOG_PASSWORDS
+	LogPasswords bool = false
+
+	// Enable logging of Http requests
+	// disabled by GOVCD_LOG_SKIP_HTTP_REQ
+	LogHttpRequest bool = true
+
+	// Enable logging of Http responses
+	// disabled by GOVCD_LOG_SKIP_HTTP_RESP
+	LogHttpResponse bool = true
+
+	// Sends log to screen. If value is either "stderr" or "err"
+	// logging will go to os.Stderr. For any other value it will
+	// go to os.Stdout
+	LogOnScreen string = ""
```

Comment:
It would be more flexible if we allowed clients to pass in their own Logger.  This would allow the SDK to integrate with whatever logging scheme the client uses.   If they don't we use the options to configure a default logger.

Reply:
Are you suggesting for us to create a Logger interface as well, so that clients can add function implementations if they are not satisfied with the default?

Reply:
Ya I think logging is one of those things where everybody has their own preferences, so allowing the client to choose how go-vcloud-director logs is probably a good thing

Reply:
@hodgesrm if we use a customized logger, we would have control only on what we do within the logging module. My purpose is also to alter the global behavior of `log.anyfunction` so that the output is the same for all the occurrence of logging within the library.
Would it be acceptable to pass instead an `io.writer`?
https://golang.org/pkg/log/#SetOutput

Reply:
On second thought, I should probably go the extra mile and use a private logger for our module. This will allow our logs to not mix with client logging. Users can change it by giving us their preferred logger.
Will be back soon with a cleaner implementation.

Reply:
OK, sounds as if you are on the case.  I also thought about the io.writer approach and even wrote it in my comment and then erased it before pressing Comment. :)  

As far as config vs. environmental variables I notice Kubernetes API clients mostly use a single environmental variable that points to a config file ($KUBECONFIG). It would be worth checking out some other Golang API clients to see what they do.

Reply:
I have reworked the logging module to use a private logger, and to accept a custom logger from users.
I have also cleaned up the library from all the occurrences of `log.Printf`, changing them to `util.GovcdLogger.Printf`.

I have not, however, introduced a configuration file. I feel that configuration files should belong in the client, not the library. We use a configuration file in our test, which is, for this purpose, a client of the library.

The next iteration on testing (already in progress) will introduce a `log` section in the test configuration file.

An updated commit will come soon.